src/query/service/src/interpreters/interpreter_list.rs

```diff
@@ -51,7 +53,23 @@ impl Interpreter for ListInterpreter {
     #[tracing::instrument(level = "debug", name = "list_interpreter_execute", skip(self), fields(ctx.id = self.ctx.get_id().as_str()))]
     async fn execute2(&self) -> Result<PipelineBuildResult> {
         let plan = &self.plan;
-        let files = list_files(&self.ctx, &plan.stage, &plan.path, &plan.pattern).await?;
+        let files = list_files(&self.ctx, &plan.stage, &plan.path).await?;
+
+        let files = if plan.pattern.is_empty() {
```

Comment:
We can add a new argument `Option<pattern>` to `list_files`

Reply:
I plan to refactor them in a stream way so that we don't need to collect all files at once.